[AY1920S1-CS2113-T16-2] El Duque

nicholasnovakovic opened the pull request:
@shaun97
@nimiew

nicholasnovakovic committed:
Optimise parser to ignore leading/trailing whitespaces in user input

nicholasnovakovic committed:
Optimise datetime input

0xKyrios committed:
Merge pull request #74 from nicholasnovakovic/master

Rework UI for date time input

0xKyrios committed:
fix bug with filtered edit command

nicholasnovakovic committed:
Merge pull request #75 from shaun97/master

fix bug with filtered edit command

nicholasnovakovic committed:
Added functionality to print undone tasks

nicholasnovakovic committed:
Optimised print output

nicholasnovakovic committed:
Optimise printing

nicholasnovakovic committed:
Optimise printing #2

sanjukta99 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,149 @@
+
+
+    public AddCommand(String description, String taskType, Optional<String> filter) {
+        this.taskType = taskType;
```
Important that you guys have JavaDoc comments for all classes. You can find the Java Coding Standards on the moduel website

sanjukta99 commented on src/main/java/duke/command/EditCommand.java:
```diff
@@ -0,0 +1,111 @@
+                }
+				if (filter == tempTaskList.get(i).getFilter()) {
+					filteredListCounter++;
+				}
```
Is the indentation for this if statement correct?

sanjukta99 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,73 @@
+        String keyword = fcArray[0];
+
+        switch (keyword) {
+            case "bye":
```
There is no indentation for case clauses in Java. Configure CheckStyle to follow this style

sanjukta99 commented on src/main/java/duke/storage/RuntimeTypeAdapterFactory.java:
```diff
@@ -0,0 +1,262 @@
+ *       .create();
+ * }</pre>
+ * Like {@code GsonBuilder}, this API supports chaining: <pre>   {@code
+ *   RuntimeTypeAdapterFactory<Shape> shapeAdapterFactory = RuntimeTypeAdapterFactory.of(Shape.class)
```
Best to delete commented out code. Makes your code more readable

sanjukta99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    public void setPriority(int i) throws DukeException {
+        switch(i) {
```
Could the getting and setting of priority be handled elsewhere? (take note of SoC)

sanjukta99 commented on src/test/java/AddCommandTest.java:
```diff
@@ -0,0 +1,17 @@
+//        assertEquals(LocalDate.of(2019, 10, 19), a.convertToLocalDate("19102019 1000"));
+//        final AddCommand b = new AddCommand("homework", "19012019 1000");
+//        assertEquals(LocalDate.of(2019, 01, 19), a.convertToLocalDate("19012019 1000"));
+    }
```
Delete commented out code

sanjukta99 commented on src/test/java/EditCommandTest.java:
```diff
@@ -0,0 +1,8 @@
+public class EditCommandTest {
+
+   // @Test
+
```
I would suggest that you guys write some test code, as and when you update the code

sanjukta99 reviewed (commented):
Take a look at my comments for some suggestions about your code. Keep in mind the Java Coding standards. 
I have noticed that most of the commands so far are the Duke commands. I hope you are working on your new features already. Feel free to speak to us if you are facing any difficulty!

nicholasnovakovic committed:
Optimise tasks for new UI handling

nicholasnovakovic committed:
Optimise AddCommand for new UI handling

nicholasnovakovic committed:
Optimise Storage package for UI handling

nimiew committed:
Merge pull request #76 from nicholasnovakovic/master

Added new print functionality

nimiew committed:
Done OffByOneChecker

nicholasnovakovic committed:
Optimisation of UI system

Optimised Parser, Command, Storage.
Deleted Deadline and ToDo classes

nimiew committed:
findCommand

nicholasnovakovic committed:
Optimise UI System

Optimised TaskList

nicholasnovakovic committed:
Optimise UI System

Optimised AddCommand

nicholasnovakovic committed:
Optimise UI System

Optimised Event

nicholasnovakovic committed:
Optimise UI System

Optimised Task

nicholasnovakovic committed:
Optimise UI System

Created DateTimeParser

nimiew committed:
Merge pull request #77 from nicholasnovakovic/master

Optimise UI interface

nimiew committed:
Merge https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
fixed edit bug

0xKyrios committed:
fixed edit bug

nicholasnovakovic committed:
Merge pull request #78 from nimiew/master

OffByOneChecker and FindCommand

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
update edit to include -t and -d

updated datetimeparser to include ddMMyy format without the time

added some exceptions to datetimeparser

0xKyrios committed:
Merge pull request #79 from shaun97/master

implemented editing for t and d

0xKyrios committed:
updated createddate for tasks

0xKyrios committed:
updated created date for task class

nimiew committed:
Merge pull request #80 from shaun97/master

updated created date for task class
NOICE

nimiew committed:
RandomCommand

nicholasnovakovic committed:
Merge pull request #81 from nimiew/master

RandomCommand

nimiew committed:
FindCommand fix mutation in place bug

0xKyrios committed:
Merge pull request #82 from nimiew/master

FindCommand fix mutation in place bug

nimiew committed:
fix OffByOneChecker due to changes in commands

nicholasnovakovic committed:
Merge pull request #83 from nimiew/master

fix OffByOneChecker due to changes in commands

nicholasnovakovic committed:
Publish new JAR release

0xKyrios committed:
implemented pomodoro command

0xKyrios committed:
Merge pull request #84 from nicholasnovakovic/master

Publish new JAR release

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nimiew committed:
implemented StartsWithChecker

0xKyrios committed:
updated split edit command into editcommand paser and editcommand

0xKyrios committed:
updated delete and done commands to work with filter

0xKyrios committed:
added addcommand parser

0xKyrios committed:
Merge pull request #86 from shaun97/master

implemented pomodoro command

0xKyrios committed:
Merge branch 'master' into master

0xKyrios committed:
Merge pull request #87 from nimiew/master

implemented StartsWithChecker

0xKyrios committed:
merge conflict resolved

0xKyrios committed:
Merge pull request #88 from shaun97/master

merge conflict resolved

0xKyrios committed:
updated checkstyle

0xKyrios committed:
updated and fix recurrence bugs

0xKyrios committed:
updated recurring task to run at startup and mark as undone if period has passed

nicholasnovakovic committed:
Merge pull request #89 from shaun97/master

updated recurring task to run at startup

nicholasnovakovic committed:
Optimised UI print output

0xKyrios committed:
implemented pomodo feature with pomo command

nicholasnovakovic committed:
Merge pull request #90 from shaun97/master

implemented pomodo feature with pomo command

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
added tests for edit command

nicholasnovakovic committed:
Create UndoStack class

nicholasnovakovic committed:
Create UndoCommand class

nicholasnovakovic committed:
Create SetCommand class

nicholasnovakovic committed:
Create InsertCommand class

nicholasnovakovic committed:
Optimise TaskList for undo functionality

nicholasnovakovic committed:
Optimise Task for undo functionality

nicholasnovakovic committed:
Update Command package to handle undo functionality

nicholasnovakovic committed:
Update DuqueParser to handle undo functionality

nicholasnovakovic committed:
Update Duke to handle undo functionality

0xKyrios committed:
Merge pull request #91 from nicholasnovakovic/master

Optimise print format and add undo function

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
udpated with test code for editcommand and pomodoro

0xKyrios committed:
added new editcommandparser test file

0xKyrios committed:
implemented checkstyle

0xKyrios committed:
fix checkstyle issues

nicholasnovakovic committed:
Merge pull request #92 from shaun97/master

fix check style issues

nicholasnovakovic committed:
Update JavaDocs for Command

nicholasnovakovic committed:
Update JavaDocs for Priority

nicholasnovakovic committed:
Update JavaDocs for Storage

nicholasnovakovic committed:
Update JavaDocs for Task, TaskList, TaskListPrinter

nimiew committed:
Merge pull request #94 from nicholasnovakovic/master

Update JavaDocs

nimiew committed:
Autoassign

nimiew committed:
Merge https://github.com/AY1920S1-CS2113-T16-2/main

nimiew committed:
javadoc

0xKyrios committed:
Merge pull request #95 from nimiew/master

AutoAssign and JavaDoc

0xKyrios committed:
updated java docs for pomodoro feature, recurrence feature

0xKyrios committed:
added tests for delete command, pomodoro, editparser and edit

0xKyrios committed:
donecommand test

nicholasnovakovic committed:
Merge pull request #96 from shaun97/master

added tests and java docs

0xKyrios committed:
Update README.md

0xKyrios committed:
Update README.md

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Rename README.md to UserGuide.adoc

0xKyrios committed:
Update README.md

0xKyrios committed:
Rename README.md to README.adoc

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update README.adoc

nimiew committed:
testing and fix bugs

nimiew committed:
Merge branch 'master' into master

nimiew committed:
Merge pull request #97 from nimiew/master

testing and fix bugs

0xKyrios committed:
fix checkstyle issues

nicholasnovakovic committed:
Add JUnit tests for InsertCommand, SetCommand, UndoCommand

0xKyrios committed:
Merge pull request #98 from nicholasnovakovic/master

Add JUnit tests for InsertCommand, SetCommand, UndoCommand

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
fix checkstyle issues

nimiew committed:
Merge pull request #101 from shaun97/master

fix checkstyle

nicholasnovakovic committed:
Optimised list print output for user

Filters for each task will now be visible to the user for their convenience

nimiew committed:
Merge pull request #104 from nicholasnovakovic/master

Optimised list print output for user

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
Delete Component Sequence.jpeg

0xKyrios committed:
Delete EditCommandSequenceDiagram-page-001.jpg

0xKyrios committed:
Add files via upload

0xKyrios committed:
Delete ModelClassDiagram.jpeg

0xKyrios committed:
Add files via upload

nicholasnovakovic committed:
Refactor package structure

nicholasnovakovic committed:
Optimise exceptions

0xKyrios committed:
Add files via upload

nicholasnovakovic committed:
Merge branch 'nicholasnovakovic-master'

# Conflicts:
#	src/main/java/duke/logic/parser/DuqueParser.java
#	src/test/java/DuqueParserTest.java
#	src/test/java/InsertCommandTest.java
#	src/test/java/ListCommandTest.java
#	src/test/java/SetCommandTest.java
#	src/test/java/UndoCommandTest.java
#	src/test/java/duke/extensions/AbnormalityCheckerTest.java
#	src/test/java/duke/logic/EditCommandParserTest.java
#	src/test/java/duke/logic/OffByOneCheckerTest.java
#	src/test/java/duke/logic/StartsWithCheckerTest.java

nimiew committed:
Merge pull request #107 from nicholasnovakovic/master

Refactor package structure

nimiew committed:
Update help command, update command names in checkers

nimiew committed:
Merge https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Merge pull request #113 from nimiew/master

Update help command

0xKyrios committed:
Add files via upload

0xKyrios committed:
Add files via upload

0xKyrios committed:
update with warning about new time period

0xKyrios committed:
Add files via upload

0xKyrios committed:
Delete UserGuide.adoc

0xKyrios committed:
Add files via upload

0xKyrios committed:
Delete [AY1920S1-CS2113-T16-2]-El-Duque-UG.pdf.pdf

0xKyrios committed:
Add files via upload

0xKyrios committed:
Delete UserGuide.adoc.pdf

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
fixed indexing bugs

0xKyrios committed:
added tests for wrong index input error
added tests for find command

0xKyrios committed:
added addcommandparser test

0xKyrios committed:
prevent doing of done task

0xKyrios committed:
fixed new line or " " command crashing program issue

0xKyrios committed:
updated to enable priority add during creating of task. updated rest of test with changes and addded test in add command parser

0xKyrios committed:
added filtered find, able to find in a particular filter

0xKyrios committed:
updated edit to include event clash checker

nicholasnovakovic committed:
Merge pull request #156 from shaun97/master

added new tests

nicholasnovakovic committed:
Update Docs and UndoTest

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Edit DG

nicholasnovakovic committed:
Optimise DG

0xKyrios committed:
implemented pomodoro joke bank

0xKyrios committed:
Merge pull request #158 from nicholasnovakovic/master

Update Documentation

0xKyrios committed:
implemented assigning of task to pomodoro task list.

0xKyrios committed:
updated pomodoro with more task functionalities

0xKyrios committed:
changed back to og filepath

nicholasnovakovic committed:
Merge pull request #159 from shaun97/master

implemented pomodoro joke bank

nimiew committed:
fix bug + added adoc

0xKyrios committed:
Merge pull request #161 from nimiew/master

fix bug + added adoc

0xKyrios committed:
fix pomodoro done bug

nicholasnovakovic committed:
Fixed bugs, transferred UG to adoc format

0xKyrios committed:
update pomodoro

0xKyrios committed:
Merge pull request #163 from nicholasnovakovic/master

Fixed bugs, transferred UG to adoc format

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
transferred UG to adocs

nicholasnovakovic committed:
Merge pull request #164 from shaun97/master

pomodoro done bug

0xKyrios committed:
updated help command

0xKyrios committed:
Merge pull request #165 from shaun97/master

updated help command

0xKyrios committed:
implemented clear command, works with filter as well

0xKyrios committed:
Added default tasks

0xKyrios committed:
updated user guide with default task creation and clear command

nicholasnovakovic committed:
Optimise program io

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Merge pull request #168 from shaun97/master

implemented default task + clear command

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
fix test bugs

nimiew committed:
Merge pull request #171 from shaun97/master

fixed test bugs

0xKyrios committed:
added a bunch of tests

0xKyrios committed:
added java docs for clear command

0xKyrios committed:
added javadocs

0xKyrios committed:
udpated UG and DG

0xKyrios committed:
updated with date parsing issue

nicholasnovakovic committed:
Merge pull request #172 from shaun97/master

added javadocs

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
fix blank string bug

0xKyrios committed:
updated print message for clear

nicholasnovakovic committed:
Merge pull request #173 from shaun97/master

fix blank string bug

nicholasnovakovic committed:
Create undo function for Clear command

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
Update README.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Optimise Undo for Clear command

nicholasnovakovic committed:
Update JavaDocs

nimiew committed:
Merge pull request #174 from nicholasnovakovic/master

Fixed bugs, optimised io, optimised undo for clear

0xKyrios committed:
updated shaun's ppp

nimiew committed:
update checkers and undo for autoassign

nimiew committed:
update userguide and ppp

nimiew committed:
update userguide autoassign

nimiew committed:
Merge pull request #175 from nimiew/master

update checkers and undo for autoassign

nimiew committed:
edit example userguide autoassign

nimiew committed:
Merge pull request #176 from nimiew/master

edit example userguide autoassign

0xKyrios committed:
updated PPP

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

# Conflicts:
#	docs/UserGuide.adoc
#	duke.json

nicholasnovakovic committed:
Update documentation

0xKyrios committed:
Merge pull request #177 from nicholasnovakovic/master

Update documentation

0xKyrios committed:
fixed formatting of user guide

0xKyrios committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

# Conflicts:
#	docs/UserGuide.adoc

0xKyrios committed:
fixed formatting issue

nicholasnovakovic committed:
Merge pull request #178 from shaun97/master

fixed merge conflicts with user guide adoc

nicholasnovakovic committed:
Update DG

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Rename [CS2113-T16-2][TeoWeiJieShaun]PPP.adoc to ShaunPPP.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
fix check style issues

0xKyrios committed:
removed failed tests

nicholasnovakovic committed:
Merge pull request #179 from shaun97/master

fix check style issues

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Update DG and UndoCommandTest

0xKyrios committed:
Merge pull request #180 from nicholasnovakovic/master

Update DG and UndoCommandTest

0xKyrios committed:
fixed checkstyle

nicholasnovakovic committed:
Updated docs

nicholasnovakovic committed:
Merge pull request #181 from shaun97/master

fixed checkstyle

nicholasnovakovic committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2113-T16-2/main

nicholasnovakovic committed:
Update AboutUs

nicholasnovakovic committed:
Update AboutUs

nicholasnovakovic committed:
Update AboutUs

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

nicholasnovakovic committed:
Edit Documentation

nicholasnovakovic committed:
Merge branch 'master' into master

0xKyrios committed:
Merge pull request #182 from nicholasnovakovic/master

Update Documentation

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc

0xKyrios committed:
Update AboutUs.adoc
